message_converter/include/converter_node.hpp

```diff
@@ -0,0 +1,202 @@
+     * Callback function when message of type libsbp_ros_msgs/MsgVelNedCov is published
+     * @param pointer to the message
+     * @return 
+     */
+    void baselineVelocityCallback(const libsbp_ros_msgs::MsgVelNedCov::ConstPtr & msg);
+
+    /**
+     * Callback function when message of type libsbp_ros_msgs/MsgPosLlh is published
+     * @param pointer to the message
+     * @return 
+     */
+    void posLLHCallback(const libsbp_ros_msgs::MsgPosLlh::ConstPtr & msg);
+
+    /**
+     * Callback function when message of type libsbp_ros_msgs/MsgBaselineNed is published
+     * @param pointer to the message
+     * @return 
+     */
+    void baselinePositionCallback(const libsbp_ros_msgs::MsgBaselineNed::ConstPtr & msg);
+
+    /**
+     * Callback function when message of type libsbp_ros_msgs/MsgGpsTime is published
+     * @param pointer to the message
+     * @return 
+     */
+    void gpsTimeCallback(const libsbp_ros_msgs::MsgGpsTime::ConstPtr & msg);
+
+    
+    // Co-ordinate conversions
+    /**
+     * Convert geodetic frame position (LLH) to NED coordinates
+     * @param LLH position, and pointer to the NED position variables
+     * @return 
+     */
+    void geodetic2ned(double latitude, double longitude, double altitude, double* north, double* east, double* down);
+
+    /**
+     * Convert geodetic frame position (LLH) to ECEF coordinates
+     * @param LLH position, and pointer to the ECEF position variables
+     * @return 
+     */
+    void geodetic2ecef(double latitude, double longitude, double altitude, double *x, double *y, double *z);
+
+    /**
+     * Convert ECEF coordinates to NED coordinates
+     * @param ECEF position, and pointer to the NED position variables
+     * @return 
+     */
+    void ecef2ned(double x_t, double y_t, double z_t, double* north, double* east, double* down);
+
+    // Setters
+    void setHomeECEF(double* x, double* y, double* z)
+    {
+        home_ecef_x_ = *x;
+        home_ecef_y_ = *y;
+        home_ecef_z_ = *z;
+    }    
+
+    // Utility functions
+    inline double deg2rad(double degrees)
+    {
+        return (degrees / 180.0) * M_PI;
+    }
+
+    inline Matrix3x3d nRe(const double lat_radians, const double lon_radians)
+    {
+        const double sLat = sin(lat_radians);
+        const double sLon = sin(lon_radians);
+        const double cLat = cos(lat_radians);
+        const double cLon = cos(lon_radians);        
+        Matrix3x3d ret;
+        ret(0, 0) = -sLat * cLon;
+        ret(0, 1) = -sLat * sLon;
+        ret(0, 2) = cLat;
+        ret(1, 0) = -sLon;
+        ret(1, 1) = cLon;
+        ret(1, 2) = 0.0;
+        ret(2, 0) = cLat * cLon;
+        ret(2, 1) = cLat * sLon;
+        ret(2, 2) = sLat;
+        return ret;
+    }
+
+    private:
+    // Timer to publish status messages at 1Hz
+    ros::Timer status_timer_;
+
+    double home_lat_, home_lon_, home_alt_;
+    double home_ecef_x_, home_ecef_y_, home_ecef_z_;
+    float home_lat_rad_, home_lon_rad_;
```

Comment:
This should be changed from `float` to `double`

Reply:
Fixed in https://github.com/Humatics/ethz_piksi_ros/pull/1/commits/48bd5c2d7ede1fbed1f91ed38d857407147ea5e8